### PR TITLE
Workflow code and documenation

### PR DESCRIPTION
**Pull request by karllark**
Will fulfill issue #114.

The workflow code and documentation for running the BEAST in "production" mode is provided.  This gives the details of how the PHAT survey (45 million stars) was fit.  The code has been generalized and tested on a METAL field.  A new example subdirectory has been added with the "production" run_beast file.

Also fixes the travis build issue by switching to python 3.6 for testing with the astropy dev version.
Also provides a fix for the case where negative values cause the log x axis 1D pdf code to fail.  Now it will just zero out the 1d pdf.   Need to fix this, but this should be done with a separate branch (see issue #120).  Right now, it just allows the code to run.

### CODE REVIEW

**Comment by heddyarab** [comment]
@karllark you'll have mine later today!

**Comment by karllark** [comment]
@heddyarab : Thanks.  Looking forward to your review.

File: beast/tools/create_source_density_map.py [hunk 1]
```diff
@@ -41,23 +36,37 @@ def make_source_dens_map(catfile,
...
-
-    written by HA 01/21/15
-    modified by KDG 09/10/15
+    FITS files written to disk.
     """
 
     cat = Table.read(catfile)
 
+    # get the columns with fluxes
+    rate_cols = [s for s in cat.colnames if s[-4:] == 'RATE']
```
Comment (galaxyumi) [inline beast/tools/create_source_density_map.py]: Isn't it too specific for HST data?

File: beast/fitting/pdf1d.py [hunk 2]
```diff
@@ -23,64 +23,77 @@ def __init__(self, gridvals, nbins, ignorebelow=None, logspacing=False):
...
-        self.max_val = tgridvals.max()
-
-        # if log spacing requested, make the transformation
-        if logspacing:
-            self.min_val = math.log10(self.min_val)
-            self.max_val = math.log10(self.max_val)
-            tgridvals = np.log10(tgridvals)
+        if len(tgridvals) <= 0:
+            self.bad = True
+            self.bin_vals = np.linspace(0.0,1.0, num=self.nbins)
```
[the remark this answers is not in the document]
Comment (galaxyumi) [reply beast/fitting/pdf1d.py]: I agree...

**Comment by galaxyumi** [comment]
Karl, please ignore my approval. I am still reviewing. Hit the wrong button and I could not find a way to cancel it. Sorry.

File: .travis.yml [hunk 3]
```diff
@@ -80,9 +80,9 @@ matrix:
           env: SETUP_CMD='build_docs -w'
 
         # Now try Astropy dev and LTS vesions with the latest 3.x and 2.7.
-        - os: linux
-          env: PYTHON_VERSION=2.7 ASTROPY_VERSION=development
-               EVENT_TYPE='pull_request push cron'
+        #- os: linux
+        #  env: PYTHON_VERSION=2.7 ASTROPY_VERSION=development
+        #       EVENT_TYPE='pull_request push cron'
```
[the remark this answers is not in the document]
Comment (karllark) [reply .travis.yml]: Good point.  I've also update to allow the development version of astropy and numpy to be allowed failures.  Really, we only care about the tests working with the stable versions.

File: beast/examples/metal_production/datamodel.py [hunk 4]
```diff
@@ -0,0 +1,297 @@
+""" Data Model interface v2.0
```
[the remark this answers is not in the document]
Comment (karllark) [reply beast/examples/metal_production/datamodel.py]: This is the sample of a "production" run that includes splitting up the observations into many subfiles.  You are correct in that we should add in the files needed to test this.  Not comfortable adding this in for the METAL data as it is not public.  Need to figure how to do this for some phat data that is not too large.  I will open an issue to solve this problem.  Do not want to hold up getting all this code in the master.

File: beast/fitting/pdf1d.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (karllark) [reply beast/fitting/pdf1d.py]: I have added a note that this is a hack.
Sensible values are not possible as this case is triggered with *all* the possible values are negative.  This happens for very low fluxes where the bias from the AST results means that all the model fluxes (physics+AST bias) are negative for the brightness range of the observations (subdivided data via the "production" workflow).

**Comment by karllark** [comment]
@galaxyumi : I have not added a docstring for the plot_indiv_fit_paper as there isn't even one for plot_indiv_fit!  We need lots more docstrings and we have an issue for this.  I added the plot_indiv_fit_paper so that we could have the formatting that was used in the paper.  The new version is more general, but the formatting is not very nice and needs fixing.

**Pull request merged**